docs/faq.md

```diff
@@ -40,3 +40,316 @@ endpoints, causing the verification to fail, and then the pod operation will fai
 The solution is to restore the kruise-controller-manager pod to a healthy state according to the prompts; or temporarily
 delete the validatingwebhookconfiguration named `kruise-validating-webhook-configuration` in the cluster, and then
 re-create it after the cluster is normal.
+
+## Workload Selection and Migration
+
+#### **Q**: When should I use CloneSet instead of Deployment?
+
+**A**: Use CloneSet when you need:
+- **In-place updates** for faster rollouts with minimal disruption
+- **Advanced scaling strategies** like specified pod deletion
+- **PVC support** for stateless workloads that need persistent storage
+- **More granular update control** with partition and maxUnavailable
+- **Better rollback capabilities** with revision management
+
+CloneSet is ideal for large-scale stateless applications where update speed and control are critical.
+
+#### **Q**: How do I migrate from Deployment to CloneSet?
+
+**A**: Follow these steps:
+1. **Backup your current Deployment** configuration
+2. **Scale down the Deployment** to 0 replicas
+3. **Create a CloneSet** with the same pod template and selector
+4. **Scale up the CloneSet** to desired replicas
+5. **Delete the old Deployment** after verification
+
```

Comment:
kruise-tools have a [command](https://github.com/openkruise/kruise-tools?tab=readme-ov-file#migrate) to migrate deployment to cloneset